tests_metricflow/integration/test_cases/itest_cumulative_metric.yaml

```diff
@@ -412,3 +412,183 @@ integration_test:
+          , {{ render_date_trunc("t.ds", TimeGranularity.QUARTER) }} AS metric_time__quarter
+          , SUM(r.revenue) AS revenue_all_time
+        FROM {{ source_schema }}.mf_time_spine t
+        INNER JOIN {{ source_schema }}.fct_revenue r ON {{ render_date_trunc("r.created_at", TimeGranularity.DAY) }} <= t.ds
+        GROUP BY
+          t.ds
+          , {{ render_date_trunc("t.ds", TimeGranularity.WEEK) }}
+          , {{ render_date_trunc("t.ds", TimeGranularity.QUARTER) }}
+      ) subq_7
+    ) subq_8
+    GROUP BY
+      metric_time__week
+      , metric_time__quarter
+      , revenue_all_time
+---
+integration_test:
+  name: window_metric_with_non_default_grains
+  description: Query a cumulative window metric with non-default grains
+  model: SIMPLE_MODEL
+  metrics: ["trailing_2_months_revenue"]
+  group_bys: ["metric_time__week", "revenue_instance__ds__month"]
+  check_query: |
+    SELECT
+      revenue_instance__ds__month
+      , metric_time__week
+      , trailing_2_months_revenue
+    FROM (
+      SELECT
+        revenue_instance__ds__month
+        , metric_time__week
+        , FIRST_VALUE(trailing_2_months_revenue) OVER (
+          PARTITION BY revenue_instance__ds__month, metric_time__week
+          ORDER BY metric_time__day
+          ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING
+        ) AS trailing_2_months_revenue
+      FROM (
+        SELECT
+          {{ render_date_trunc("t.ds", TimeGranularity.MONTH) }} AS revenue_instance__ds__month
+          , t.ds AS metric_time__day
+          , {{ render_date_trunc("t.ds", TimeGranularity.WEEK) }} AS metric_time__week
+          , SUM(r.revenue) AS trailing_2_months_revenue
+        FROM {{ source_schema }}.mf_time_spine t
+        INNER JOIN {{ source_schema }}.fct_revenue r
+        ON
+          (
+            {{ render_date_trunc("r.created_at", TimeGranularity.DAY) }} <= t.ds
+          ) AND (
+            {{ render_date_trunc("r.created_at", TimeGranularity.DAY) }} > {{ render_date_sub("t", "ds", 2, TimeGranularity.MONTH) }}
+          )
+        GROUP BY
+          {{ render_date_trunc("t.ds", TimeGranularity.MONTH) }}
+          , t.ds
+          , {{ render_date_trunc("t.ds", TimeGranularity.WEEK) }}
+      ) subq_7
+    ) subq_8
+    GROUP BY
+      revenue_instance__ds__month
+      , metric_time__week
+      , trailing_2_months_revenue
+---
+integration_test:
+  name: grain_to_date_metric_with_non_default_grains
+  description: Query a cumulative grain to date metric with non-default grains
+  model: SIMPLE_MODEL
+  metrics: ["revenue_mtd"]
+  group_bys: ["revenue_instance__ds__quarter", "revenue_instance__ds__year"]
+  check_query: |
+    SELECT
+      revenue_instance__ds__quarter
+      , revenue_instance__ds__year
+      , revenue_mtd
+    FROM (
+      SELECT
+        revenue_instance__ds__quarter
+        , revenue_instance__ds__year
+        , FIRST_VALUE(revenue_mtd) OVER (
```

Comment:
Oh this is interesting. If it's month to date and people truncate to year that's super strange, right? It's just always month to date for January or whatever? I guess the answer is "don't do that" - but I do wonder if we should have a different default from FIRST_VALUE - I guess people can eventually override it to LAST_VALUE (which might be more sensible).